server/plugin.go

```diff
@@ -310,6 +315,13 @@ func (p *Plugin) start(isRestart bool) {
 	if err = p.API.RegisterCommand(p.createCommand()); err != nil {
 		p.API.LogError("Failed to register command", "error", err)
 	}
+
+	p.cancelKeyFuncLock.Lock()
+	if !isRestart && p.cancelKeyFunc == nil {
+		p.tabAppJWTKeyFunc, p.cancelKeyFunc = setupJWKSet()
+	}
+	p.cancelKeyFuncLock.Unlock()
```

Comment:
There is a fair amount of code behind the `setupJWKSet` call, that goes into some third party libs (e.g. https://github.com/MicahParks/jwkset).  If there is a panic the HTTP handler will recover but this mutex won't release and further requests will block.  

```suggestion
	defer p.cancelKeyFuncLock.Unlock()
```

Reply:
I think if `setupJWKSet` panics, the plugin itself will shutdown since `start` and in turn `OnActivate()` / `OnConfigurationChange` will panic and kill the process.